### DISCUSSION TITLE
Make不能通过静态检查

### DISCUSSION THREAD

**Discussion by taynpg** [chenshuo/muduo#629] [Q&A]
-- Build files have been written to: /home/ubuntu/code/build/release-cpp11
+ make install
Scanning dependencies of target muduo_base
[  0%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/AsyncLogging.cc.o
[  0%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Condition.cc.o
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/CountDownLatch.cc.o
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/CurrentThread.cc.o
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Date.cc.o
[  1%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Exception.cc.o
[  2%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/FileUtil.cc.o
[  2%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/LogFile.cc.o
[  2%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/Logging.cc.o
[  3%] Building CXX object muduo/base/CMakeFiles/muduo_base.dir/LogStream.cc.o
/home/ubuntu/code/muduo-2.0.2/muduo/base/LogStream.cc: In member function ‘void muduo::LogStream::staticCheck()’:
/home/ubuntu/code/muduo-2.0.2/muduo/base/LogStream.cc:228:38: error: static assertion failed: kMaxNumericSize is large enough
  228 |   static_assert(kMaxNumericSize - 10 > std::numeric_limits<long double>::digits10,
      |                 ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [muduo/base/CMakeFiles/muduo_base.dir/build.make:180: muduo/base/CMakeFiles/muduo_base.dir/LogStream.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1223: muduo/base/CMakeFiles/muduo_base.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

我应该如何做才能通过检查呢？
编译环境：
ubuntu@primary:~/code/muduo-2.0.2$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/aarch64-linux-gnu/9/lto-wrapper
Target: aarch64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=aarch64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libquadmath --disable-libquadmath-support --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --enable-fix-cortex-a53-843419 --disable-werror --enable-checking=release --build=aarch64-linux-gnu --host=aarch64-linux-gnu --target=aarch64-linux-gnu
Thread model: posix
gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)

**Comment by chenshuo** [comment]
```sh
$ git show 8303763edb0ab9e9e47f0d07c0432e2fb150d72c
commit 8303763edb0ab9e9e47f0d07c0432e2fb150d72c
Author: Shuo Chen <chenshuo@chenshuo.com>
Date:   2021-07-12 21:00:22 +0000

    fix build on Raspberry Pi 4 where std::numeric_limits<long long>::digits10 == 33, instead of 18 on x86.

diff --git a/muduo/base/LogStream.h b/muduo/base/LogStream.h
index a28eb46..442b468 100644
--- a/muduo/base/LogStream.h
+++ b/muduo/base/LogStream.h
@@ -165,7 +165,7 @@ class LogStream : noncopyable
 
   Buffer buffer_;
 
-  static const int kMaxNumericSize = 32;
+  static const int kMaxNumericSize = 48;
 };
 
 class Fmt // : noncopyable
```

**Reply by taynpg** [reply]
OK
